File: 1376. Time Needed to Inform All Employees/Solution.c
#include <math.h>

int dfs(int i, int* manage, int* informTime)
{
    if (manage[i] != -1) {
        informTime[i] += dfs(manage[i], manage, informTime);
        manage[i] = -1;
    }
    return informTime[i];
}

int numOfMinutes(int n, int headID, int* manager, int managerSize, int* informTime, int informTimeSize)
{
    int res = 0;
    for (int i = 0; i < n; i++)
        res = fmax(res, dfs(i, manager, informTime));
    return res;
}

File: 1376. Time Needed to Inform All Employees/Solution.java
class Solution {
    private int dfs(int i, int[] manager, int[] informTime) {
        if (manager[i] != -1) {
            informTime[i] += dfs(manager[i], manager, informTime);
            manager[i] = -1;
        }
        return informTime[i];
    }

    public int numOfMinutes(int n, int headID, int[] manager, int[] informTime) {
        int res = 0;
        for (int i = 0; i < n; i++) {
            res = Math.max(res, dfs(i, manager, informTime));
        }
        return res;
    }
}
